common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -52,13 +54,23 @@
 
     private final String mAuthType;
 
-    AuthTypes(String authType) {
+    AuthType(String authType) {
       mAuthType = authType;
     }
 
     public String getAuthName() {
       return mAuthType;
     }
+
+    /**
+     * Validate the authentication type string and return a corresponding AuthType.
+     * @param authTypeStr authentication type string from configuration
+     * @return the corresponding AuthType
+     * @throws java.lang.IllegalArgumentException if the string does not match any type
+     */
+    public static AuthType getValidAuthType(String authTypeStr) {
+      return AuthType.valueOf(authTypeStr.toUpperCase(Locale.ENGLISH));
```

Comment:
NIT: for debugging purpose, maybe let's catch the exception and throw it again so we can add a more specific error message? (I just don't know what error message would be for this `java.lang.IllegalArgumentException` here. If you think it's good enough, don't bother to fix.)